demosplan/DemosPlanCoreBundle/Controller/Segment/DraftsInfoController.php

```diff
@@ -106,8 +108,8 @@ public function editAction(
         } catch (StatementNotFoundException $e) {
             $this->getMessageBag()->add('error', 'error.statement.not.found');
             throw $e;
-        } catch (StatementAlreadySegmentedException $e) {
-            $this->getMessageBag()->add('error', 'error.statement.already.segmented');
+        } catch (StatementAlreadySegmentedException|LockedByAssignmentException $e) {
+            $this->messageBag->add('error', $e->getMessage());
```

Comment:
Is it safe to directly output the Exception message here?

Reply:
I would say so, yes. They both can only be thrown in line 93 `$segmentableStatementValidator->validate($statementId);` and the message generated there is the old explicit one for the `StatementAlreadySegmentedException` and a user friendly one for `LockedByAssignmentException` stating that a user can't edit a statement that is not assigned to them. I think that convey relevant information without any technical details.

Reply:
OK for now, but I feel very uncomfortable about passing Exception messages directly to user. What if a new implementation does not know that it is passed? Perhaps we should introduce something to improve DX that makes it clear that this data is passed to users directly

Reply:
Great, thx, :heart: